modules/util/error.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+import (
+	"io/fs"
+)
+
+// Common Errors forming the base of our error system
+//
+// Many Errors returned by Gitea can be tested against these errors
+// using errors.Is.
+var (
+	ErrInvalid    = fs.ErrInvalid    // "invalid argument"
+	ErrPermission = fs.ErrPermission // "permission denied"
+	ErrExist      = fs.ErrExist      // "file already exists"
+	ErrNotExist   = fs.ErrNotExist   // "file does not exist"
```

Comment:
Hmm, if we already redeclare these errors we might as well give them their actual name
(`ErrInvalidArgument`, `ErrPermissionDenied`, `ErrAlreadyExist`, `ErrNotExist`).
While they are a bit longer they are also more clear and require less thought process what they could mean.

Reply:
So I think if we rename these we should actually not have them equal to `fs.ErrInvalid`.

Reply:
Why do you mean?

Reply:
So part of this is that we could say just decide that this `util.ErrInvalid` is unnecessary and we should just use `fs.ErrInvalid` and it's sort of advertising that they are equivalent so that it's a bit more obvious that `errors.Is(util.ErrInvalid, fs.ErrInvalid)` but yeah I'm open to making them not equivalent.

Reply:
well, now my suggestion applies again.

Reply:
done